var/spack/repos/builtin/packages/gpi-space/package.py

```diff
@@ -49,7 +49,7 @@ class GpiSpace(CMakePackage):
                type=("build", "run"))
     depends_on("pkgconf",
                type="build")
-    depends_on("boost@1.62.0:1.63.0 +coroutine +context cxxstd=14")
+    depends_on("boost@1.62.0:1.63.0 +atomic +chrono +coroutine +context +date_time +filesystem +iostreams +program_options +random +regex +serialization +test +timer cxxstd=14")
```

Comment:
In the GPI-Space repository, when you execute

```
git grep -h '<boost/' | sed 's,# *include ,,g;s,<boost/\([^/\.]*\)[/\.].*,\1,' | sort | uniq
```

the result is

```
algorithm
align
any
archive
asio
bimap
blank
current_function
enable_shared_from_this
filesystem
format
functional
io
iostreams
iterator
lambda
lexical_cast
make_shared
mpl
msm
optional
preprocessor
program_options
random
range
ref
serialization
shared_ptr
system
test
thread
tokenizer
utility
uuid
variant
version
```

From that list I do not see the dependency to ` atomic`, ` chrono`, `date_time`,`regex` and `timer`. There is a dependency to ` system` and `thread` though. -> I know that `asio` causes the dependency to `coroutine`  and `context`, so my question is:

How do you came up with the list of dependencies? Is there a way to automatically get/maintain it?

Reply:
The dependency on `atomic` and `regex` I derived from the error messages. Here a little example, there are many more:
```
  >> 651    CMake Error at cmake/include/util-cmake/add_macros.cmake:390 (add_executable):
     652      Target "gspc-bootstrap-rifd" links to target "Boost::atomic" but the target
     653      was not found.  Perhaps a find_package() call is missing for an IMPORTED
     654      target, or an ALIAS target is missing?
     655    Call Stack (most recent call first):
     656      modules/gspc_add_executable.cmake:54 (extended_add_executable)
     657      src/rif/CMakeLists.txt:91 (gspc_add_executable)
     658    
     659    
  >> 660    CMake Error at cmake/include/util-cmake/add_macros.cmake:390 (add_executable):
     661      Target "gspc-bootstrap-rifd" links to target "Boost::regex" but the target
     662      was not found.  Perhaps a find_package() call is missing for an IMPORTED
     663      target, or an ALIAS target is missing?
     664    Call Stack (most recent call first):
     665      modules/gspc_add_executable.cmake:54 (extended_add_executable)
     666      src/rif/CMakeLists.txt:91 (gspc_add_executable)
  ```

The others I took from the Boost components requested by GPI-Space's CMakelists and matched them against the available variants in the Boost package.

```
find_package (Boost 1.61.0 REQUIRED COMPONENTS
  chrono
  date_time
  filesystem
  iostreams
  program_options
  random
  serialization
  system
  test_exec_monitor
  thread
  timer
  unit_test_framework
)
```

Reply:
Thanks, I see. So... Where does the dependency to ` Boost::atomic`  or ` Boost::regex`  comes from? When doing

```
git grep -h Boost:: | tr ' ' '\n' | tr -d ')' | grep -v ^$ | grep ^Boost:: | sort | uniq
```

the result is

```
Boost::base
Boost::boost
Boost::date_time
Boost::filesystem
Boost::iostreams
Boost::program_options
Boost::serialization
Boost::system
Boost::thread
Boost::unit_test_framework
```

and direct search (in the repository and its sub-repositories) leads nothing

```
src> grep Boost::atomic * -R
src> grep Boost::regex * -R
src> 
```

I want to make sure the list of dependencies does is precise.

Reply:
A lot of these dependencies are pulled in by `FindBoost.cmake`:

- `regex` is a dependency of `iostreams`
- `atomic`, `chrono`, and `date_time` are dependencies of `thread`
- ...

There are a few more of these transitive dependencies which can be seen here: https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L1199

Reply:
So long story short, I am afraid we actually depend on all of these.
At least up to the current release of GPI-Space.